### ISSUE TITLE
Handle mongouri

### ISSUE THREAD

**Issue by joeimer** [MongoEngine/flask-mongoengine#254]
Hi

With the new mongoengine and pymongo version I got an error.
The problem was, that flask_mongoengine handles the mongo uri too now.

After I changed the connection.py it worked fine for me.
I just removed the # Handle uri style connections part

```
    """
    # Handle uri style connections
    if host.startswith('mongodb://'):
        uri_dict = uri_parser.parse_uri(host)
        if uri_dict['database']:
            resolved['host'] = uri_dict['database']
        if uri_dict['password']:
            resolved['password'] = uri_dict['password']
        if uri_dict['username']:
            resolved['username'] = uri_dict['username']
        if uri_dict['options'] and uri_dict['options']['replicaset']:
            resolved['replicaSet'] = uri_dict['options']['replicaset']
    """
```

**Comment by losintikfos** [comment]
@joeimer can you make a pull request for this?

Ta!

**Comment by llvtt** [comment]
This appears related to https://github.com/MongoEngine/flask-mongoengine/issues/247.

**Comment by lafrech** [comment]
Contributors willing to fix the connexion regression in 0.8 are welcome to join in https://github.com/MongoEngine/flask-mongoengine/issues/71#issuecomment-252554167.

**Comment by Folyd** [comment]
Silly bug, still unfixed?
    
    if uri_dict['database']:
            resolved['host'] = uri_dict['database']

**Comment by wojcikstefan** [comment]
Should be fixed in v0.8.1, v0.8.2, and v0.9.0.
